### ISSUE TITLE
Support grouping services

### ISSUE THREAD

**Issue by TwinProduction** [TwinProduction/gatus#13]
Supporting service groups could allow a cuter front end experience.

i.e.
```yaml
services:
  - name: k8s-cluster-watch-dog
    url: http://k8s-cluster-watch-dog-v1.tools-${ENVIRONMENT}:8080/health
    group: core         <-------
    interval: 1m
    conditions:
      - "[STATUS] == 200"
      - "[BODY].status == UP"
  - name: prometheus
    url: http://prometheus-operator-prometheus.kube-system:9090/-/healthy
    group: core         <-------
    interval: 1m
    conditions:
      - "[STATUS] == 200"
      - "[BODY] == Prometheus is Healthy."
```

would generate a dashboard that puts both `k8s-cluster-watch-dog` and `prometheus` under the "core" folder.

![image](https://user-images.githubusercontent.com/15699766/93811861-c7d40200-fc1e-11ea-8fe7-6747616f1993.png)

Could also support tags, and allow filtering by tags instead

i.e.
```yaml
services:
  - name: k8s-cluster-watch-dog
    url: http://k8s-cluster-watch-dog-v1.tools-${ENVIRONMENT}:8080/health
    tags:              <-------
      - core
    interval: 1m
    conditions:
      - "[STATUS] == 200"
      - "[BODY].status == UP"
  - name: prometheus
    url: http://prometheus-operator-prometheus.kube-system:9090/-/healthy
    tags:              <-------
      - core
      - metrics
    interval: 1m
    conditions:
      - "[STATUS] == 200"
      - "[BODY] == Prometheus is Healthy."
```

![image](https://user-images.githubusercontent.com/15699766/93812350-74ae7f00-fc1f-11ea-9ca0-7ce12f0f634c.png)

Forgive the terrible drafts, just thought of this on the fly.

**Comment by hurnhu** [comment]
@TwinProduction Would you be open to using a front end framework like Vuejs?

**Comment by TwinProduction** [comment]
> @TwinProduction Would you be open to using a front end framework like Vuejs?
> It would allow for easy additions like filtering on tags.

@hurnhu Honestly, at the moment, everything is read-only on the UI and rather simple, so I feel that using frameworks is a bit too _heavy_, in fact, that's why I didn't include popper.js and created a custom tooltip instead. Even filtering without a custom framework doesn't require that much work.

That said, perhaps when there's enough features to warrant a full-fledged framework, Vuejs would be a great candidate.

**Comment by dieterdemeyer** [comment]
@TwinProduction ,

We're using this framework extensively at the moment to monitor several endpoints towards our Kubernetes clusters because it's a great tool btw.

We are also a fan of improving the layout of the dashboard to be able to group several checks together.
I've played around with the index.html file but I'm not a frontend expert.
If we would try another attempt at implementing this feature, could you give a few pointers on where to start ?
It shouldn't be too much work to implement given that the api call gives all results back and could include a group attribute, correct ?

Thanks in advance.

**Comment by TwinProduction** [comment]
@dieterdemeyer Actually, you're in luck.

I started working on this yesterday 🤣 

This is what I have so far.
![image](https://user-images.githubusercontent.com/15699766/100383866-add2e900-2fec-11eb-94d6-714e747c3944.png)

I'm not completely sure about the design yet, but I'd love a second opinion if you can.

I'm working right now, but I should be able to work on it tonight.

**Comment by dieterdemeyer** [comment]
@TwinProduction Ah, even better 😄 

Already looks pretty good.
Some things that come to mind when I think more about the design are as follows:

1. Being able to minimize the group would be nice. We have a lot of endpoints to monitor so being able to only focus on or see a specific group while the rest is minimized could be a good addition.
2. This might be more difficult to implement but to have a green or red check mark (same icon as for the endpoints) next to the group name to indicate that one of the endpoints within that group is failing.
3. Something that would benefit from the previous item is to include an option to minimize all the groups by default. That way, you can have several “functional” groups and instantly see which group had a failing endpoint.

Not sure if all options are that easy to implement but what do you think ?

**Comment by TwinProduction** [comment]
1 was already on my agenda, as for 2, I came up with a lazier approach for now: if there's 0 failures, show a checkmark, otherwise, show a yellow `~`.

![Gatus with some groups minimized](https://user-images.githubusercontent.com/15699766/100394714-2ac18b00-300c-11eb-8f97-20b044bbf943.png)
![Gatus with all groups minimized](https://user-images.githubusercontent.com/15699766/100394728-34e38980-300c-11eb-9457-445ca494b6d1.png)

Regarding 3, I'm also thinking about it, but not yet.
I'm planning on adding a `frontend` configuration in the YAML file at some point specifically to tweak settings like that, but the first feature I'll add is the ability to only show the toolbox on click rather than on hover. Accidentally hovering over a status has been annoying me more than it should have honestly.

FYI, this is the tooltip I'm talking about: 
![Gatus tooltip](https://user-images.githubusercontent.com/15699766/100394918-d4a11780-300c-11eb-982f-b699cfe3a807.png)

Thank you for the feedback, by the way :)


> 
> 
> @TwinProduction Ah, even better 😄
> 
> Already looks pretty good.
> Some things that come to mind when I think more about the design are as follows:
> 
>   1. Being able to minimize the group would be nice. We have a lot of endpoints to monitor so being able to only focus on or see a specific group while the rest is minimized could be a good addition.
>   2. This might be more difficult to implement but to have a green or red check mark (same icon as for the endpoints) next to the group name to indicate that one of the endpoints within that group is failing.
>   3. Something that would benefit from the previous item is to include an option to minimize all the groups by default. That way, you can have several “functional” groups and instantly see which group had a failing endpoint.
> 
> 
> Not sure if all options are that easy to implement but what do you think ?

1 was already on my agenda, as for 2, I came up with a lazier approach for now: if there's 0 failures, show a checkmark, otherwise, show a yellow `~`.

![Gatus with some groups minimized](https://user-images.githubusercontent.com/15699766/100394714-2ac18b00-300c-11eb-8f97-20b044bbf943.png)
![Gatus with all groups minimized](https://user-images.githubusercontent.com/15699766/100394728-34e38980-300c-11eb-9457-445ca494b6d1.png)

Regarding 3, I'm also thinking about it, but not yet.

First, I want to add a parameter to toggle whether to show the tooltip on click or on hover. 
Accidentally hovering over a status has been annoying me more than it should have.

I see two potential ways of doing this, and I'm not entirely sure which route to take yet:

1. Adding a `frontend` configuration in the YAML file at some point specifically to tweak settings like that, but the first feature 
2. Exposing these configurations only on the frontend and let each individual users configure it however they prefer. Persistence would be done with `localStorage`.

FYI, this is the tooltip I'm talking about: 
![Gatus tooltip](https://user-images.githubusercontent.com/15699766/100394918-d4a11780-300c-11eb-982f-b699cfe3a807.png)

Thank you for the feedback, by the way :)

**Comment by dieterdemeyer** [comment]
@TwinProduction , no problem 😄 
Can't wait to use the new version with this feature included.

FWIW, regarding the tooltip, I would also go with letting the user decide what they want to configure based on their preference..

**Comment by TwinProduction** [comment]
@dieterdemeyer The feature has been merged into latest. 
Could you give it a try and let me know how it goes before I release it in v1.9.0?

**Comment by dieterdemeyer** [comment]
@TwinProduction , definitely. I can test it next Monday and I'll let you know.

**Comment by dieterdemeyer** [comment]
@TwinProduction , just finished testing and it works.
The only issue that I found is that the service name can't be the same when it's defined in 2 groups.
I have a group "API" and a group "Legacy API" where a service name is defined with the same name.
But the dashboard only shows the service from the "API" group.
When I rename the service so that it's unique across the configuration file, it does show up in the dashboard..

**Comment by TwinProduction** [comment]
@dieterdemeyer Mind giving it another try? Should be resolved now.

**Comment by dieterdemeyer** [comment]
@TwinProduction , just did some tests and it works. 👍

**Comment by TwinProduction** [comment]
Released in [v1.9.0](https://github.com/TwinProduction/gatus/releases/tag/v1.9.0)
